runtime/compiler/control/CompilationThread.cpp

```diff
@@ -480,10 +480,22 @@ int32_t TR::CompilationInfo::computeDynamicDumbInlinerBytecodeSizeCutoff(TR::Opt
 TR_YesNoMaybe TR::CompilationInfo::shouldActivateNewCompThread()
    {
 #if defined(J9VM_OPT_CRIU_SUPPORT)
-   // Don't activate any threads until the restore if the threads should be suspended for checkpoint
+   // Only allow a single comp thread pre-checkpoint if -XX:+DebugOnRestore
```

Comment:
Could you please make sure that we don't encounter a situation where, for whatever reason, we suspend all compilation threads and then we never activate even a single one because of this code?
I would actually check the number of comp threads that are active, and if there is none, then allow one to get activated.

Reply:
Updated in force push.